verification/getCertificateChain_test.go

```diff
@@ -0,0 +1,172 @@
+// Licensed under the Apache-2.0 license
+
+package verification
+
+import (
+	"crypto/x509"
+	"encoding/pem"
+	"fmt"
+	"log"
+	"testing"
+
+	zx509 "github.com/zmap/zcrypto/x509"
+	zlint "github.com/zmap/zlint/v3"
+	"github.com/zmap/zlint/v3/lint"
+)
+
+// This file is used to test the Get Certificate Chain command by using a simulator/emulator
+
+func TestGetCertificateChain(t *testing.T) {
+	support_needed := []string{"AutoInit", "X509"}
+	instance, err := GetTestTarget(support_needed)
+	if err != nil {
+		if err.Error() == "Requested support is not supported in the emulator" {
+			t.Skipf("Warning: Failed executing TestGetCertificateChain command due to unsupported request. Hence, skipping the command execution")
+		} else {
+			log.Fatal(err)
+		}
+	}
+	testGetCertificateChain(instance, t)
+}
+
+func TestGetCertificateChain_SimulationMode(t *testing.T) {
+
+	support_needed := []string{"AutoInit", "Simulation", "X509"}
+	instance, err := GetTestTarget(support_needed)
+	if err != nil {
+		if err.Error() == "Requested support is not supported in the emulator" {
+			t.Skipf("Warning: Failed executing TestGetCertificateChain_SimulationMode command due to unsupported request. Hence, skipping the command execution")
+		} else {
+			log.Fatal(err)
+		}
+	}
+	testGetCertificateChain(instance, t)
+}
+
+func testGetCertificateChain(d TestDPEInstance, t *testing.T) {
+	if d.HasPowerControl() {
+		err := d.PowerOn()
+		if err != nil {
+			log.Fatal(err)
+		}
+		defer d.PowerOff()
+	}
+	client, err := NewClient256(d)
+	if err != nil {
+		t.Fatalf("Could not initialize client: %v", err)
+	}
+
+	// Initialize request input parameters
+	getCertificateChainReq := GetCertificateChainReq{
+		Offset: 0,
+		Size:   MAX_CHUNK_SIZE,
+	}
+
+	getCertificateChainResp, err := client.GetCertificateChain(&getCertificateChainReq)
+	if err != nil {
+		t.Fatalf("Could not get Certificate Chain: %v", err)
+	}
+
+	//fmt.Println(getCertificateChainResp)
+	checkCertificateChain(t, getCertificateChainResp.CertificateChain)
+
+	// TODO: When DeriveChild is implemented, call it here to add more TCIs and call GetCertificateChain again.
```

Comment:
This isn't relevant to GetCertificateChain in the same way it is for CertifyKey

Reply:
Did you mean the comment? Yes I removed it.